activities/resolution_hierarchy.py

```diff
@@ -18,31 +18,72 @@
 from bossutils.multidimensional import XYZ, ceildiv
 from bossutils.multidimensional import range as xyz_range
 
-from heaviside.activities import fanout
+import types
+import json
+import time
+import random
+from multiprocessing import Pool, Value, cpu_count
+from datetime import timedelta, datetime
 
 log = logger.BossLogger().logger
 
-# Fanout variables, see heaviside.activities.fanout for full details
-# int - seconds: The delay between launch subprocesses and polling for status
-POLL_DELAY = 0 # DP NOTE: STATUS_DELAY * MAX_NUM_PROCESSES should provide enough delay
-               #          when polling for status
+######################################################
+### Configure Remote Debugging via SIGUSR1 trigger ###
+def sigusr_handler(sig, frame):
+    try:
+        import rpdb2
+        rpdb2.start_embedded_debugger('password')
+    except:
+        log.exception("SIGUSR1 caught but embedded debugger could not be started")
 
-# int - seconds: The inner delay between status queries
-#                Helps limit API speed, so as to not run into throttling issues
-STATUS_DELAY = 1
+def enable_debug_handler():
+    import signal
+    signal.signal(signal.SIGUSR1, sigusr_handler)
+######################################################
 
-# int: Maximum number of concurrent subprocess executions to have running
-MAX_NUM_PROCESSES = 50
 
-# int - seconds: The initial rampup delay to allow AWS resources to scale
-RAMPUP_DELAY = 0 # DP NOTE: Don't need to rampup for downsample
+###########################################################
+###               Configuration Options                 ###
+### Modify these values to modify the activity behavior ###
 
-# float: The backoff value to multiple RAMPUP_DELAY by for each subprocess launch
-#        When RAMPUP_DELAY is zero there is no longer a delay between launched
-RAMPUP_BACKOFF = 0.8
+# int: The number of volumes each downsample_volume lambda should downsample
+BUCKET_SIZE = 8
 
-# int: The number of volumes each sub_sfn should downsample with each execution
-BUCKET_SIZE = 10
+# datetime.delta: The runtime of the downsample_volume lambda
+MAX_LAMBDA_TIME = timedelta(seconds=120)
+
+# int: The number of status poll cycles that have the same count for more
+#      lambdas to be launched (total time is UNCHANGING_LAUNCH * MAX_LAMBDA_TIME)
+UNCHANGING_LAUNCH = 3
+
+# int: The number of status poll cycles that have the same count for the
+#      activity to check the lambda throttle count and wait until throttling
+#      is no longer effecting the cubes queue count
+UNCHANGING_THROTTLE = 4
+
+# int: The number of status poll cycles that have to be the same for an
+#      exception to be raised
+#      This is a fail safe so the activity doesn't hang if there is a problem
+UNCHANGING_MAX = 5
+
+# int: The number of multiprocessing pool workers to use
+POOL_SIZE = int(cpu_count() * 2)
```

Comment:
Just curious, does this actually have to cast to int?

Reply:
I think it needs to be a whole number, it is passed to `multiprocessing.Pool`. I was previously multiplying by 2.5 and wanted to make sure to not have a float result.